src/datareaders/lucid/lucid_reader.py

```diff
@@ -0,0 +1,86 @@
+from src.datareaders.resources import get_data_resource
+from src.datareaders.table_enumerations import Sources
+from src.datareaders.database_connection import DatabaseConnection
+from src.datareaders.lucid.lucid_parser import LucidParser
+from src.datareaders.data_object_holders import Point, PointType
+from json import load as json_load
+
+
+class LucidReader:
+    """TODO: Write docs for function. Since I'm not sure what form this is going to
+       take, we'll come back to this..."""
+
+    def __init__(self, file_name, source):
+        self.file_path = get_data_resource("csv_files/" + file_name)
+        self.source = source  # Use the enumeration for Lucid
+        self.db_connection = DatabaseConnection()  # Start up connection to DB
+        self.lucid_parser = LucidParser()  # initialize parser class for looking at CSVs
+        self.lucid_parser.read_csv(self.file_path)  # Load CSV
+        self.lucid_parser.create_point_identities()  # Create values to insert into Points Table
+        self.lucid_parser.create_point_values()  # Create values to insert into PointValue Table
+
+    def add_to_db(self):
+        """
+        Adds the points and point values from the loaded CSV file into the database.
+
+        :return: None
+        """
+
+        self.add_points()
+        self.add_point_values()
+
+    def add_points(self):
+        """
+        Adds the points to the points table.
+
+        :return: None
+        """
+
+        successfully_inserted = []
+        unsuccessfully_inserted = []
+        for point in self.lucid_parser.point_identities:
+            try:
+                self.db_connection.add_unique_building(point.building)
+                self.db_connection.add_unique_room(None, point.building)
```

Comment:
Because of the table construction we do need to add a None room for all buildings that we are adding.  It might not need to be there, but it will need to be added and we don't necessarily have building information without the point information

Reply:
What do you mean by "this"? We need a dummy room for lucid meters since a meter/point has to have a room ID, not a building ID. We might want to make it more similar to Siemens though where the dummy room is called <building>_dummy_room or <building>_room rather than NULL.

Reply:
None has been changed to building_dummy_room to keep things consistent

Reply:
Understand that we need a dummy room. I read 
for point in self.lucid_parser.point_identities
as for each point we are adding a dummy room in that points building. Which makes me concerned since a lot of points will be in the same building, so we're essentially "overwriting" (or whatever add_unique actually does under the hood) the dummy room and the building a lot. Which obviously wouldn't be ideal from a performance perspective. Or am I misunderstanding the contents of self.lucid_parser.point_identities?